### PR TITLE
implement audit logs in gaius_common

### PR DESCRIPTION
**Pull request by vaibhavac**

### CODE REVIEW

File: gaius_common/middleware/changeLog.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    return _request_ctx.get()
+
+def add_to_request_context(key, value):
+    context = _request_ctx.get()
+    context[key] = value
+    _request_ctx.set(context)
+
+def get_ip_address():
+    try:
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
```
Comment (suryadana) [inline gaius_common/middleware/changeLog.py]: this one will be show host ip address not client IP address

**Review by suryadana — CHANGES_REQUESTED** [review]

File: gaius_common/middleware/changeLog.py (hunk 1 above)
Comment (vaibhavac) [reply gaius_common/middleware/changeLog.py]: The host IP address is now showing correctly.

File: gaius_common/middleware/changeLog.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+def add_to_request_context(key, value):
+    context = _request_ctx.get()
+    context[key] = value
+    _request_ctx.set(context)
+
+_ip_cache = None
+_IP_CHECK_URL = 'https://ifconfig.me/ip'
+
+def get_ip_address():
```
Comment (suryadana) [inline gaius_common/middleware/changeLog.py]: This method same like previous

**Review by suryadana — CHANGES_REQUESTED** [review]

File: gaius_common/middleware/changeLog.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+def get_current_request():
+    return _request_ctx.get()
+
+def add_to_request_context(key, value):
+    context = _request_ctx.get()
+    context[key] = value
+    _request_ctx.set(context)
+
+def get_ip_address(request):
+    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR')
```
Comment (suryadana) [inline gaius_common/middleware/changeLog.py:20]: add x-real-ip also at the first

File: gaius_common/common/document.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    new_value = Text()
+    timestamp = Date()
+    type = Keyword()
+    hostname = Text()
+    api_endpoint = Text()
+    user = Text()
+    ip_address = Ip()
+
+    class Index:
+        name = 'change-logs-staging-v3'
```
Comment (suryadana) [inline gaius_common/common/document.py:19]: make it as environment variable

**Review by suryadana — CHANGES_REQUESTED** [review]

File: gaius_common/middleware/changeLog.py (hunk 3 above)
Comment (vaibhavac) [reply gaius_common/middleware/changeLog.py:20]: This is fixed. I've added x-real-ip at the first.

**Review by suryadana — APPROVED** [review]

**Pull request merged**